src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditMessage.java

```diff
@@ -339,15 +335,11 @@ void addRestRequestInfo(final RestRequest request, final AuditConfig.Filter filt
 
     public void addTransportHeaders(Map<String,String> headers, boolean excludeSensitiveHeaders) {
         if(headers != null && !headers.isEmpty()) {
-            if(excludeSensitiveHeaders) {
-                final Map<String,String> headersClone = new HashMap<String,String>(headers)
-                        .entrySet().stream()
-                        .filter(map -> !map.getKey().equalsIgnoreCase(AUTHORIZATION_HEADER))
-                        .collect(Collectors.toMap(p -> p.getKey(), p -> p.getValue()));
-                auditInfo.put(TRANSPORT_REQUEST_HEADERS, headersClone);
-            } else {
-                auditInfo.put(TRANSPORT_REQUEST_HEADERS, new HashMap<String,String>(headers));
+            final Map<String, String> headersClone = new HashMap<>(headers);
```

Comment:
Avoid code duplication. Merge `addTransportHeaders` and `addRestHeaders` to a single method with one more parameter.

Reply:
Both are accepting different map types.

Reply:
Accept `Map<String, ?>`, value type actually does not matter.

Reply:
Not in favor of merging `addTransportHeaders` and `addRestHeaders` into one function with accepting generics as value types will be permissive for consumers. 
Maybe the validation and filtering can be done as a separate function with generics. 
Intention of this PR was not to do a lot of refactor (check PR body). Maybe raise a separate issue?

Reply:
Yes, please open a follow up PR.
- avoid code duplication. IMO, it is OK to accept generics as AuditMessage does not care in what format Headers are provided.
- header filtering logic should avoid creating a new map and then removing elements from them.

Reply:
Create #657